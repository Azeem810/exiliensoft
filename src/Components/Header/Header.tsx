import { Wrapper } from "../../StyledComponets/HeaderStyle";
import Logo from "../../assets/images/logo.png";
import { Link } from "react-router-dom";
import { useNavigate } from "react-router-dom";
const Header = () => {
  const navigate = useNavigate();
  const handleHome = () => {
    navigate("/");
  };
  const handleContact = () => {
    navigate("/contact");
  };
  return (
    <Wrapper>
      <div className="logo" onClick={handleHome}>
        <img src={Logo} alt="Logo " />
        <p>Digital agency</p>
      </div>
      <nav>
        <ul>
          <li>
            <Link to={"/about"}>About</Link>
          </li>
          <li>
            <Link to={"/services"}>Services</Link>
          </li>
          <li>
            <Link to={"/projects"}>Project</Link>
          </li>
        </ul>
        <button onClick={handleContact}>Get In Touch</button>
      </nav>
    </Wrapper>
  );
};

export default Header;
