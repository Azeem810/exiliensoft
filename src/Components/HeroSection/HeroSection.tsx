import {
  HeroSectionCompany,
  HeroSectionWrapper,
} from "../../StyledComponets/HeroSectionStyle";
import HeroImage from "../../assets/images/hero-section-image.png";
import ProjectImage from "../../assets/images/projects.png";
import CustomerLogos from "../../assets/images/customer-logos.png";
import Services from "../Services/Services";
import Testimonial from "../Testimonial/Testimonial";
import Blogs from "../Blogs/Blogs";
const HeroSection = () => {
  return (
    <>
      <HeroSectionWrapper>
        <div>
          <h1>
            Building Brands in the <span>Digital Age</span>
          </h1>
          <p>
            Your partner in navigating the ever-evolving landscape of digital
            marketing. From conceptualization to execution, we craft tailored
            solutions that drive results and elevate your brand to new heights.
          </p>
          <button>Learn More</button>
        </div>
        <div className="hero-image-container">
          <img
            src={HeroImage}
            alt="Hero section image"
            className="hero-image"
          />
          <img
            src={ProjectImage}
            alt="Project image"
            className="project-image"
          />
        </div>
      </HeroSectionWrapper>
      <HeroSectionCompany>
        <div>
          <p>Powering next-gen companies</p>

          <img src={CustomerLogos} alt="customers logos" />
        </div>
      </HeroSectionCompany>
      <Services />
      <Testimonial />
      <Blogs />
    </>
  );
};

export default HeroSection;
