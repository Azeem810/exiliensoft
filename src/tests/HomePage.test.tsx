import { renderWithProviders } from "./Wrapper";
import HeroSection from "../Components/HeroSection/HeroSection";
import { fireEvent } from "@testing-library/react";
import Header from "../Components/Header/Header";
import { useNavigate } from "react-router-dom";


const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe("Home Page tests", () => {

  it("initial snapshot", () => {
    const container = renderWithProviders(<HeroSection />);
    expect(container).toMatchSnapshot();
  });


  it("Check Learn more how many times is coming in the document", () => {
    const container = renderWithProviders(<HeroSection />);

    expect(container.getAllByText("Learn More")).toHaveLength(2);
  });

  

  it("Should navigate to contact page", () => {
    
    const container = renderWithProviders(<Header />);
    fireEvent.click(container.getByText(/Get In Touch/));
    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith('/contact');
  });
  
});


