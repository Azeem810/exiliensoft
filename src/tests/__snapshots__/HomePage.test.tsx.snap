// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page tests initial snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-bdfCDU kECChr"
      >
        <div>
          <h1>
            Building Brands in the 
            <span>
              Digital Age
            </span>
          </h1>
          <p>
            Your partner in navigating the ever-evolving landscape of digital marketing. From conceptualization to execution, we craft tailored solutions that drive results and elevate your brand to new heights.
          </p>
          <button>
            Learn More
          </button>
        </div>
        <div
          class="hero-image-container"
        >
          <img
            alt="Hero section image"
            class="hero-image"
            src="hero-section-image.png"
          />
          <img
            alt="Project image"
            class="project-image"
            src="projects.png"
          />
        </div>
      </section>
      <div
        class="sc-gsTDqH imMVaM"
      >
        <div>
          <p>
            Powering next-gen companies
          </p>
          <img
            alt="customers logos"
            src="customer-logos.png"
          />
        </div>
      </div>
      <section
        class="sc-dlfmHC eDwHNo"
      >
        <div>
          <h2>
            Explore Our Services
          </h2>
          <h3>
            We are self-service data analytics software that lets you create visually.
          </h3>
          <div
            class="card-container"
          >
            <div
              class="card"
            >
              <img
                alt="App Development"
                src="app-development.png"
              />
              <h4>
                App Development
              </h4>
              <p>
                ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
              </p>
            </div>
            <div
              class="card"
            >
              <img
                alt="Web Designing"
                src="we-designing.png"
              />
              <h4>
                Web Designing
              </h4>
              <p>
                ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
              </p>
            </div>
            <div
              class="card"
            >
              <img
                alt="Graphic Designing"
                src="graphic-designing.png"
              />
              <h4>
                Graphic Designing
              </h4>
              <p>
                ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
              </p>
            </div>
            <div
              class="card"
            >
              <img
                alt="Digital Marketing"
                src="digital-marketing.png"
              />
              <h4>
                Digital Marketing
              </h4>
              <p>
                ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
              </p>
            </div>
          </div>
          <div
            class="btn"
          >
            <button>
              Learn More
            </button>
          </div>
        </div>
      </section>
      <section
        class="sc-hKgKIp kbguKQ"
      >
        <div
          class="image-container"
        >
          <span
            class="image-text"
          >
            “
          </span>
          <img
            alt="Testimonial Image"
            class="hero-image"
            src="testimonial-image.png"
          />
        </div>
        <div
          class="container"
        >
          <p
            class="client-message"
          >
            “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be.”
          </p>
          <p
            class="about-client"
          >
            <span>
              David Calathan-
            </span>
             Director of Design Operations, New York
          </p>
          <div
            class="next-previous-icons"
          >
            <img
              alt="Previous image"
              src="previous.png"
            />
            <img
              alt="Next image"
              src="next.png"
            />
          </div>
        </div>
      </section>
      <section
        class="sc-eCstZk drQjFP"
      >
        <div
          class="main-container"
        >
          <h6>
            Our Blogs
          </h6>
          <div
            class="card-container"
          >
            <div>
              <img
                alt="blog 1 image"
                class="image-style"
                src="blog1.png"
              />
              <div
                class="card-text"
              >
                <p
                  style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
                >
                  Realtime analytics
                </p>
                <p>
                  “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
                </p>
                <img
                  alt=" next image"
                  class="next-image-style"
                  src="next.png"
                />
              </div>
            </div>
            <div>
              <img
                alt="blog 1 image"
                class="image-style"
                src="blog2.png"
              />
              <div
                class="card-text"
              >
                <p
                  style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
                >
                  Realtime analytics
                </p>
                <p>
                  “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
                </p>
                <img
                  alt=" next image"
                  class="next-image-style"
                  src="next.png"
                />
              </div>
            </div>
            <div>
              <img
                alt="blog 1 image"
                class="image-style"
                src="blog3.png"
              />
              <div
                class="card-text"
              >
                <p
                  style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
                >
                  Realtime analytics
                </p>
                <p>
                  “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
                </p>
                <img
                  alt=" next image"
                  class="next-image-style"
                  src="next.png"
                />
              </div>
            </div>
          </div>
          <div
            class="btn"
          >
            <button>
              View all posts
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-bdfCDU kECChr"
    >
      <div>
        <h1>
          Building Brands in the 
          <span>
            Digital Age
          </span>
        </h1>
        <p>
          Your partner in navigating the ever-evolving landscape of digital marketing. From conceptualization to execution, we craft tailored solutions that drive results and elevate your brand to new heights.
        </p>
        <button>
          Learn More
        </button>
      </div>
      <div
        class="hero-image-container"
      >
        <img
          alt="Hero section image"
          class="hero-image"
          src="hero-section-image.png"
        />
        <img
          alt="Project image"
          class="project-image"
          src="projects.png"
        />
      </div>
    </section>
    <div
      class="sc-gsTDqH imMVaM"
    >
      <div>
        <p>
          Powering next-gen companies
        </p>
        <img
          alt="customers logos"
          src="customer-logos.png"
        />
      </div>
    </div>
    <section
      class="sc-dlfmHC eDwHNo"
    >
      <div>
        <h2>
          Explore Our Services
        </h2>
        <h3>
          We are self-service data analytics software that lets you create visually.
        </h3>
        <div
          class="card-container"
        >
          <div
            class="card"
          >
            <img
              alt="App Development"
              src="app-development.png"
            />
            <h4>
              App Development
            </h4>
            <p>
              ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
            </p>
          </div>
          <div
            class="card"
          >
            <img
              alt="Web Designing"
              src="we-designing.png"
            />
            <h4>
              Web Designing
            </h4>
            <p>
              ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
            </p>
          </div>
          <div
            class="card"
          >
            <img
              alt="Graphic Designing"
              src="graphic-designing.png"
            />
            <h4>
              Graphic Designing
            </h4>
            <p>
              ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
            </p>
          </div>
          <div
            class="card"
          >
            <img
              alt="Digital Marketing"
              src="digital-marketing.png"
            />
            <h4>
              Digital Marketing
            </h4>
            <p>
              ipsum dolor sit amet, consectetur adipiscing elit. Nullam euismod metus vel sem bibendum, a bibendum justo tempor. Sed vel lectus
            </p>
          </div>
        </div>
        <div
          class="btn"
        >
          <button>
            Learn More
          </button>
        </div>
      </div>
    </section>
    <section
      class="sc-hKgKIp kbguKQ"
    >
      <div
        class="image-container"
      >
        <span
          class="image-text"
        >
          “
        </span>
        <img
          alt="Testimonial Image"
          class="hero-image"
          src="testimonial-image.png"
        />
      </div>
      <div
        class="container"
      >
        <p
          class="client-message"
        >
          “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be.”
        </p>
        <p
          class="about-client"
        >
          <span>
            David Calathan-
          </span>
           Director of Design Operations, New York
        </p>
        <div
          class="next-previous-icons"
        >
          <img
            alt="Previous image"
            src="previous.png"
          />
          <img
            alt="Next image"
            src="next.png"
          />
        </div>
      </div>
    </section>
    <section
      class="sc-eCstZk drQjFP"
    >
      <div
        class="main-container"
      >
        <h6>
          Our Blogs
        </h6>
        <div
          class="card-container"
        >
          <div>
            <img
              alt="blog 1 image"
              class="image-style"
              src="blog1.png"
            />
            <div
              class="card-text"
            >
              <p
                style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
              >
                Realtime analytics
              </p>
              <p>
                “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
              </p>
              <img
                alt=" next image"
                class="next-image-style"
                src="next.png"
              />
            </div>
          </div>
          <div>
            <img
              alt="blog 1 image"
              class="image-style"
              src="blog2.png"
            />
            <div
              class="card-text"
            >
              <p
                style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
              >
                Realtime analytics
              </p>
              <p>
                “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
              </p>
              <img
                alt=" next image"
                class="next-image-style"
                src="next.png"
              />
            </div>
          </div>
          <div>
            <img
              alt="blog 1 image"
              class="image-style"
              src="blog3.png"
            />
            <div
              class="card-text"
            >
              <p
                style="font-size: 24px; font-weight: 500; color: rgb(15, 0, 73); margin-bottom: 20px;"
              >
                Realtime analytics
              </p>
              <p>
                “The results have been incredible. With Power Digital, it feels like they’re in our trench, supporting and understanding us. They’re like a partner and mentor in helping us get where we want to be. “The results have been incredible.
              </p>
              <img
                alt=" next image"
                class="next-image-style"
                src="next.png"
              />
            </div>
          </div>
        </div>
        <div
          class="btn"
        >
          <button>
            View all posts
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
